src/Features/Core/Portable/CodeFixes/Suppression/AbstractSuppressionBatchFixAllProvider.cs

```diff
@@ -0,0 +1,621 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
this is broken out from the original batch fixer.  The suppression fixers do so much customizing of the batch fixer that i prefer to just extract this out using the original logic so that there's no risk of breaking supression batch fixing.

Reply:
@mavasani  for thoughts here.  Suppression seems very complex, so this just carves it out so we play with it independently from everything else.  This also allows us to keep the DocumentBasedFixAllProvider very clean and simple (including no facilities for overriding).

Reply:
I had thought about such a refactoring but it was leading to lot of code duplication, as you would have noticed. Is the code being added here being duplicated elsewhere? I will go through the remaining changes in the PR as well

Reply:
Not sure I totally get your question. The basic idea is that there are three ways to do fix all:

The per document way (went in yesterday).
The batch fixer way (which shares code with per-document). That is this pr.
The suppression way. This is independent. That is this pr as well.

Reply:
I've tried to make the first two (per document, and batch) much simpler and saner.

Reply:
My question: Does this lead to code duplication between suppression fix all batch fixer and well known batch fixer (latter is publically exposed)?

Reply:
no.  they are now almost entirely independent.  the only sharing is how they get diagnostics.  but hte processing o diagnostics/applicatoin is unique to each.